crates/apollo-mcp-server/src/operations.rs

```diff
@@ -161,18 +163,53 @@ pub enum MutationMode {
     All,
 }
 
-#[derive(Debug, Clone, Serialize)]
+#[derive(Debug, Clone)]
 pub struct RawOperation {
     source_text: String,
-    #[serde(skip_serializing_if = "Option::is_none")]
     persisted_query_id: Option<String>,
+    headers: Option<HeaderMap<HeaderValue>>,
+    variables: Option<HashMap<String, Value>>,
+}
+
+// Custom Serialize implementation for RawOperation
```

Comment:
```suggestion
// Custom Serialize implementation for RawOperation
// This is needed because reqwests HeaderMap/HeaderValue/HeaderName don't derive Serialize
```